enterprise/api/v1/views.py

```diff
@@ -510,3 +515,94 @@ class EnterpriseCustomerReportingConfigurationViewSet(EnterpriseReadOnlyModelVie
     )
     filter_fields = FIELDS
     ordering_fields = FIELDS
+
+
+class CouponCodesView(APIView):
+    """
+    API to request coupon codes.
+    """
+    permission_classes = (permissions.IsAuthenticated,)
+    authentication_classes = (JwtAuthentication, BearerAuthentication, SessionAuthentication,)
+    throttle_classes = (ServiceUserThrottle,)
+
+    REQUIRED_PARAM_EMAIL = 'email'
+    REQUIRED_PARAM_ENTERPRISE_NAME = 'enterprise_name'
+    OPTIONAL_PARAM_NUMBER_OF_CODES = 'number_of_codes'
+
+    MISSING_REQUIRED_PARAMS_MSG = "Some required parameter(s) missing: {}"
+
+    def get_required_query_params(self, request):
+        """
+        Gets ``email``, ``enterprise_name``, and ``number_of_codes``,
+        which are the relevant parameters for this API endpoint.
+
+        :param request: The request to this endpoint.
+        :return: The ``email``, ``enterprise_name``, and ``number_of_codes`` from the request.
+        """
+        email = get_request_value(request, self.REQUIRED_PARAM_EMAIL, '')
+        enterprise_name = get_request_value(request, self.REQUIRED_PARAM_ENTERPRISE_NAME, '')
+        number_of_codes = get_request_value(request, self.OPTIONAL_PARAM_NUMBER_OF_CODES, '')
+        if not (email and enterprise_name):
+            raise CodesAPIRequestError(
+                self.get_missing_params_message([
+                    ("'email'", bool(email)),
+                    ("'enterprise_name'", bool(enterprise_name)),
+                ])
+            )
+        return email, enterprise_name, number_of_codes
+
+    def get_missing_params_message(self, parameter_state):
+        """
+        Get a user-friendly message indicating a missing parameter for the API endpoint.
+        """
+        params = ', '.join(name for name, present in parameter_state if not present)
+        return self.MISSING_REQUIRED_PARAMS_MSG.format(params)
+
+    def post(self, request):
+        """
+        POST /enterprise/api/v1/request_codes
+
+        Requires a JSON object of the following format:
+        >>> {
+        >>>     "email": "bob@alice.com",
+        >>>     "enterprise_name": "IBM",
+        >>>     "number_of_codes": "50"
+        >>> }
+
+        Keys:
+        *email*
+            Email of the customer who has requested more codes.
+        *enterprise_name*
+            The name of the enterprise requesting more codes.
+        *number_of_codes*
+            The number of codes requested.
+        """
+        try:
+            email, enterprise_name, number_of_codes = self.get_required_query_params(request)
+        except CodesAPIRequestError as invalid_request:
+            return Response({'error': str(invalid_request)}, status=HTTP_400_BAD_REQUEST)
+
+        subject_line = _('Code Management - Request for more Codes by {token_enterprise_name}').format(
```

Comment:
nit: The word "more" is unnecessary in the subject line: "`Code Management - Request for Codes by {token_enterprise_name}`"